### PR TITLE
ds/partition: Add reverse DNS prefix

### LINKED ISSUE #16639: ds/partition: Add service_prefix to aws_partition
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

`com.amazonaws` is included in service names in the standard partition. However, in non-standard partitions, this service prefix can be different such as `cn.com.amazonaws` in China. The `aws_partition` data source should provide this service prefix.

### New or Affected Resource(s)

* aws_partition

### Potential Terraform Configuration

```hcl
data "aws_region" "current" {}

data "aws_partition" "current" {}

resource "aws_vpc_endpoint" "test" {
  vpc_id       = aws_vpc.test.id
  service_name = "${data.aws_partition.current.service_prefix}.${data.aws_region.current.name}.s3"
}
```

### References

* aws/aws-cdk#9864
* #16521 ([comment](https://github.com/hashicorp/terraform-provider-aws/pull/16521#discussion_r534172378))

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #16639

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
data-source/aws_partition: Add `reverse_dns_prefix` attribute (#16638)
```

### Acceptance Tests in GovCloud (`us-gov-west-1`)

```
--- PASS: TestAccAWSPartition_basic (14.28s)
```

### Acceptance Tests in `us-west-2`

```
--- PASS: TestAccAWSPartition_basic (11.16s)
```

### Acceptance Tests in `us-east-1`

```
--- PASS: TestAccAWSPartition_basic (10.02s)
```

### CODE REVIEW

File: aws/data_source_aws_partition.go [hunk 1]
```diff
@@ -15,10 +17,16 @@ func dataSourceAwsPartition() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
 			"dns_suffix": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"service_prefix": {
```
Comment (bflad) [inline aws/data_source_aws_partition.go:26]: Let's call this `reverse_dns_prefix` for clarity. 👍

**Review by bflad — DISMISSED** [review]
Otherwise, this is probably good to go 😄

File: aws/data_source_aws_partition.go (hunk 1 above)
Comment (YakDriver) [reply aws/data_source_aws_partition.go]: Done

**Comment by YakDriver** [comment]
Tests after changes:

GovCloud:
```
--- PASS: TestAccAWSPartition_basic (11.72s)
```

`us-west-2`:
```
--- PASS: TestAccAWSPartition_basic (8.01s)
```

**Review by bflad — APPROVED** [review]
Looks good to me 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSPartition_basic (8.69s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSPartition_basic (15.10s)
```

**Comment by bharaths** [comment]
This doesn't work correctly in AWS China. It returns "com.cn.amazonaws" while "cn.com.amazonaws" is the expected value

**Comment by bflad** [comment]
Yes it would. 😅  Thank you for the report, will open an issue and fix this.

**Pull request merged**